react/components/QuantityStepper.tsx

```diff
@@ -0,0 +1,201 @@
+    id: 'store/product-list.quantityStepper.decreaseQuantity',
+  },
+})
+
+interface Props {
+  id?: string
+  value?: number
+  maxValue?: number
+  onChange?: (value: number) => void
+  disabled?: boolean
+  unitMultiplier?: number
+  measurementUnit?: string
+}
+
+const QuantityStepper: VFC<Props> = ({
+  id,
+  value = 1,
+  maxValue,
+  onChange,
+  disabled,
+  unitMultiplier = 1,
+  measurementUnit = 'un',
+}) => {
+  const intl = useIntl()
+
+  const normalizedValue = normalizeValue(value, maxValue)
+
+  const [getUpdatedValue] = useQuantitySelectorState({
+    maxValue,
+    measurementUnit,
+  })
+
+  const [focused, setFocused] = useState(false)
+
+  const [currentValue, setCurrentValue] = useState(
+    intl.formatNumber(
+      parseDisplayValue({
+        value: normalizedValue.toString(),
+        maxValue,
+        unitMultiplier,
+      }),
+      { useGrouping: false }
+    )
+  )
+
+  const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = (
+    evt
+  ) => {
+    setCurrentValue(evt.target.value)
+  }
+
+  const handleInputBlur = () => {
+    const { validatedValue, validatedDisplayValue } = getUpdatedValue({
+      value: currentValue,
+      unitMultiplier,
+    })
+
+    onChange?.(validatedValue)
+
+    setCurrentValue(validatedDisplayValue)
+    setFocused(false)
+  }
+
+  const handleDecrease: React.MouseEventHandler = (evt) => {
+    evt.preventDefault()
+
+    const parsedValue = parseDisplayValue({
+      value: normalizedValue.toString(),
+      maxValue,
+      unitMultiplier,
+    })
+
+    const { validatedValue, validatedDisplayValue } = getUpdatedValue({
+      value: (parsedValue - unitMultiplier).toString(),
+      unitMultiplier,
+    })
+
+    onChange?.(validatedValue)
+    setCurrentValue(validatedDisplayValue)
+  }
+
+  const handleIncrease: React.MouseEventHandler = (evt) => {
+    evt.preventDefault()
+
+    const parsedValue = parseDisplayValue({
+      value: normalizedValue.toString(),
+      maxValue,
+      unitMultiplier,
+    })
+
+    const { validatedValue, validatedDisplayValue } = getUpdatedValue({
+      value: (parsedValue + unitMultiplier).toString(),
+      unitMultiplier,
+    })
+
+    onChange?.(validatedValue)
+    setCurrentValue(validatedDisplayValue)
+  }
+
+  useEffect(() => {
+    if (focused) {
+      return
+    }
+
+    setCurrentValue(
+      intl.formatNumber(
+        parseDisplayValue({
+          value: normalizedValue.toString(),
+          maxValue,
+          unitMultiplier,
+        }),
+        { useGrouping: false }
+      )
+    )
+  }, [focused, intl, normalizedValue, maxValue, unitMultiplier])
+
+  const uniqueId = id ? `product-list-quantity-stepper-${id}` : undefined
+
+  return (
+    <div className="flex">
+      <button
+        className={classnames(
+          'pa4 ba br2 br--left flex items-center justify-center',
+          {
+            'c-muted-1 b--muted-4 hover-b--muted-3 bg-muted-5 hover-bg-muted-4 pointer': !disabled,
```

Comment:
Why is this styling different from the increment button style (line 187)? I noticed that the left side is greyer and it looks disabled, so I'm not sure if that's intentional. Also, could you link the figma reference?

![image](https://user-images.githubusercontent.com/26108090/101530040-22ecd980-3970-11eb-8a4b-bd67a01d7206.png)

Reply:
it is meant to be like that, [here is the figma file](https://www.figma.com/file/eO8KLGSxYr6N8VQq9JcoJD/Checkout-IO---Ongoing---Cart?node-id=186%3A149316&viewport=656%2C304%2C0.0496190786361694340

Reply:
Ohh, it's the prototype section, right?

Reply:
no, it is the one inside the "proposition" section:

<img width="922" alt="image" src="https://user-images.githubusercontent.com/10223856/101665862-47a68700-3a2c-11eb-984d-2fa1f7da675b.png">